annorxiver_modules/document_helper.py

```diff
@@ -115,6 +136,102 @@ def generate_doc_vector(model, document_path, xpath, filter_tags=filter_tag_list
     return []
 
 
+def generate_doc_vector_parallel(
+    model, xpath, document_queue, vector_queue, filter_tags=filter_tag_list
+):
+    """
+    This method is designed to construct document vectors for a given xml document.
+    Every document has specific tags that are striped in order to have accurate embeddings
+
+    Keyword arguments:
+        model - the model to extract word vectors from
+        xpath_str - the xpath string to extract tags from the xml document
+        filter_tag_list - the list of tags to strip from the xml document
+    """
+    while True:
+
+        document_path = document_queue.get()
+
+        if document_path is None:
+            break
+
+        word_vectors = []
+
+        tree = ET.parse(open(document_path[2], "rb"), parser=parser)
+
+        # Process xml without specified tags
+        ET.strip_tags(tree, *filter_tags)
+
+        root = tree.getroot()
+        all_text = root.xpath(xpath)
+        all_text = list(map(lambda x: "".join(list(x.itertext())), all_text))
+        all_text = " ".join(all_text)
+
+        all_tokens = list(
+            map(
+                lambda x: x.lemma_,
+                filter(
+                    lambda tok: tok.lemma_ in model.wv
+                    and tok.lemma_ not in nlp.Defaults.stop_words,
+                    nlp(all_text),
+                ),
+            )
+        )
+
+        word_vectors += [model.wv[text] for text in all_tokens]
+
+        # skips weird documents that don't contain text
```

Comment:
Are these weird documents those very old ones that are images and not text? Have you run into other strange cases - i'm thinking about those older papers that published the protein sequences within the text of the paper for instance. I guess these would just be outliers. I'm not sure if it would throw off your embedding

Reply:
>Are these weird documents those very old ones that are images and not text?

You are correct. Those weird documents are just image scans of old papers focused on DNA/protein sequences. They don't affect results as they would appear as empty vectors that I'm skipping.